daemons/scheduled-ci-build/app.py

```diff
@@ -14,33 +15,39 @@
 app = domovoi.Domovoi(configure_logs=False)
 
 
-def define_test_type(test_type: str, envvar: str, schedule_expression: str):
-    @app.scheduled_function(schedule_expression)
-    def integration_test(event, context):
+def define_build(branch_name: str, build_type: str, schedule_expression: str, **env):
+    def f(event, context):
         travis_token = os.environ["TRAVIS_TOKEN"]
         rule_name = event["resources"][0].split(":")[-1]
         body = {
             'request': {
-                'branch': "master",
-                'message': f"{test_type} test started by {context.function_name} from {rule_name}",
+                'branch': branch_name,
+                'message': f"{build_type} build started by {context.function_name} from {rule_name}",
                 'config': {
                     'merge_mode': "deep_merge",
                     'env': {
-                        'matrix': [f"{envvar}=1"]
+                        'matrix': [f"{k}={v}" for k, v in env.items()]
                     },
                 },
-            }}
+            }
+        }
         headers = {
             'Accept': "application/json",
             'Travis-API-Version': "3",
             'Authorization': f"token {travis_token}",
         }
-
         return requests.post(
             "https://api.travis-ci.org/repo/HumanCellAtlas%2Fdata-store/requests",
             headers=headers,
             json=body,
         ).json()
 
+    f.__name__ = 'scheduled_ci_build_' + hashlib.sha1((branch_name + build_type).encode('utf8')).hexdigest()[:8]
+    globals()[f.__name__] = app.scheduled_function(schedule_expression)(f)
```

Comment:
This is too clever. I don't think declaring two separate daemons should require this level of tinkering. What's wrong with simply defining two app entry points and having them both call out to a shared utility function that does the work?

Reply:
I tried to preserve @ttung's original intent of using a factory (and be clever). But I'll change it.

Reply:
Just to clarify: *I* wanted to be clever. I agree that @kislyuk's suggestion is preferable.

Reply:
Thanks, this looks great :)